codegen/core/src/main/java/software/amazon/smithy/python/codegen/generators/UnionGenerator.java

```diff
@@ -125,9 +130,12 @@ raise NotImplementedError()
 
                 """, unknownSymbol.getName());
         memberNames.add(unknownSymbol.getName());
+        writer.popState();
 
-        writer.write("type $L = $L\n", parentName, String.join(" | ", memberNames));
+        writer.pushState(new UnionSection(shape, parentName, memberNames));
+        writer.write("$L = Union[$L]\n", parentName, String.join(" | ", memberNames));
```

Comment:
Need to revert to the older union syntax; autodoc doesn't natively support the new syntax yet: https://github.com/sphinx-doc/sphinx/issues/10785

Reply:
Can you add this reference as a comment in the code? That context will get lost I guarantee